test/services/SubmissionModelServiceSpec.scala

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import config.AppConfig
+import generators.MessageGenerators
+import models.{TraderWithEori, TraderWithoutEori}
+import models.messages.NormalNotification
+import models.messages.request.{InterchangeControlReference, _}
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.Gen
+import org.scalatest.{FreeSpec, MustMatchers}
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import play.api.inject.Injector
+
+class SubmissionModelServiceSpec extends FreeSpec with MustMatchers with GuiceOneAppPerSuite with MessageGenerators with ScalaCheckDrivenPropertyChecks {
+
+  def injector: Injector = app.injector
+
+  def appConfig: AppConfig = injector.instanceOf[AppConfig]
+
+  val convertToSubmissionModel = new SubmissionModelService(appConfig)
+
+  "SubmissionModelService" - {
+
+    "must convert NormalNotification to ArrivalNotificationRequest for traders with Eori" in {
+
+      val notifications: Gen[(ArrivalNotificationRequest, NormalNotification)] = {
+        for {
+          arrivalNotificationRequest <- arbitraryArrivalNotificationRequestWithEori.arbitrary
+        } yield {
+
+          val dateTime = arrivalNotificationRequest.meta.interchangeControlReference.dateTime
+          val normalNotification: NormalNotification = {
+            NormalNotification(
+              movementReferenceNumber = arrivalNotificationRequest.header.movementReferenceNumber,
+              notificationPlace = arrivalNotificationRequest.header.arrivalNotificationPlace,
+              notificationDate = dateTime.toLocalDate,
+              customsSubPlace = arrivalNotificationRequest.header.customsSubPlace,
+              trader = TraderWithEori(
+                name = arrivalNotificationRequest.traderDestination.name,
+                streetAndNumber = arrivalNotificationRequest.traderDestination.streetAndNumber,
+                postCode = arrivalNotificationRequest.traderDestination.postCode,
+                city = arrivalNotificationRequest.traderDestination.city,
+                countryCode = arrivalNotificationRequest.traderDestination.countryCode,
+                eori = arrivalNotificationRequest.traderDestination.eori.get
```

Comment:
Can we use the `OptionValue` trait and use the `.value` instead of `.get`? It gives much better errors for tests
```suggestion
                eori = arrivalNotificationRequest.traderDestination.eori.value
```

Reply:
I think there are a few more in the tests that this can be done for also